### PR TITLE
Add a "rural" biome in mod form

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Mods "Adds a 'rural' biome in mod form, ready for testing for mainline mods."```

#### Purpose of change
I've been saying for ages we need to have a few different biomes expressing just population density: at least rural and dense urban, as well as the current small town / suburban. As well, changes I'm prepping to autolearn will severely nerf "innawoods" wilderness-only games. I see these problems as related - the ability to play in a wilderness biome with some places to loot, but far fewer, is one way (not the only way I'm working on) to help preserve the wilderness playstyle, as well as testing out biomes.

#### Describe the solution
I attempted to create a pure wilderness biome, but the inability to adjust appearance rates of overmap specials in `region_map_settings.json` means I can't reduce the appearance rates of farms and cabins.

The final result is really good as a rural biome, though, which is one of my targets. It's full of farms and small towns with just a few shops. There are no mansions or labs.

Remaining flaws are that I can't readily mess with the appearance rates of specials, as mentioned above, and that the vehicle groups for the tiny villages spawning are not appropriate, made for cities rather than little rural villages. As well, if this is to be made into a biome rather than a mod that affects the whole world, it will need some kind of json attribute for regional_map_settings to control city spacing and size.

#### Describe alternatives you've considered
As a mod, I could do some individual tweaks to vehicle groups and map specials, but I'm not really trying to make a mod here... that's just how I'm getting to the objective of finding what needs to work better for biomes.

#### Testing
image

#### Additional context
I'll put out a couple Issues for the things I'm missing here at some point.

### CODE REVIEW

File: data/json/overmap/overmap_special/specials.json [hunk 1]
```diff
@@ -601,7 +605,7 @@
     "city_distance": [ 5, -1 ],
     "city_sizes": [ 1, -1 ],
     "occurrences": [ 5, 100 ],
-    "flags": [ "UNIQUE" ]
+    "flags": [ "CLASSIC", "FARM" ]
```
Comment (Night-Pryanik) [inline data/json/overmap/overmap_special/specials.json]: Please don't remove the `UNIQUE` flag. With this flag, the chance to spawn will be the intended 5 out of 100. Without this flag, this farm will spawn minimum 5 and maximum 100 times on a single overmap.

Comment (I-am-Erk) [reply data/json/overmap/overmap_special/specials.json]: That looks like a copy-paste error, it shouldn't be getting removed anywhere

**Comment by mkikt4743** [comment]
4 houses seems rather small even for a small town, and gas stations and other 'rest stop' areas should be generated more in these small towns, since thos are common in small towns along major roads. Also, no labs is severe, and I think it'd make more sense to have labs out in rural areas rather than near cities, since there is those rooms with beds in them implying that some of the scientists lived there. From a lore perspective, it's a lot easier for the government to hide something if it's hard to just stumble across when you're biking on the road, and less people are around to discover it.

**Comment by I-am-Erk** [comment]
You're misunderstanding both the purpose and limitations of this. If I enable labs, I enable all labs at the current frequency, so it doesn't work like that. Since I intend this as an intermediate form of innawoods play, labs and mansions totally break it. When added to the main game the biome might eventually have a lab once in a rare while, but they do need to be getting a lot rarer so I'm not in a rush.

I have a similar lack of control over town size, as mentioned in the pr. If you want more houses, set the city size bigger. Gas stations and rest stops are both among the possible spawns, with gas stations being by a good margin the most likely thing to appear in a town.

**Comment by scorpion451** [comment]
One input I'll give as someone who lives in a rural area, that sounds like the sort of long-term procedural generation ideas you're looking to spark:

In small towns and the countryside, things tend to aggregate around a lot of things that would be specials in the map gen. 

Say, a steelmill or mine operation might have a cluster of commercial sites and houses that grew up around it over the years- employees building houses nearby, diners opening up to cater to people on lunch break, small grocery stores to grab milk on the way home, etc. 

Same thing still happens with places like research labs (which they often do put in the middle of nowhere for safety and privacy reasons), factories, and megastores meant to serve numerous towns, that sort of thing. 

A practical example that could be simple to implement using the nested generation stuff ya'll have been playing with are highway towns. Basically they pop up when you have a large farm or ranch sell off the high-dollar "quaint summer house"("last gas station for 20 miles", "Fine Antiques and Canned Pecans!", etc) land right next to the road, but keep all of the land behind that, now with a buffer between them and the highway. You'll even see this go full circle now and then when a town expands to encompass an agricultural area- full fledged barns-and-tractors acreage surrounded by a ring of what looks like a subdivision or strip malls, with one narrow lot dedicated to the driveway back into the center property. In game terms, you make an agricultural site one back from a road with a dirt road connection, then have a chance to spawn houses and commercial sites where the farm is adjacent to a road.

TL;DR: Things cluster around big stuff in the country, this might help make interesting templates both here and in other biomes.

**Comment by I-am-Erk** [comment]
Yes, I was thinking in general about the possibility of adding a mine or a mill as a "shop" in rural mapgen settings, but with current mapgen that still makes a mine appear inside the city surrounded by houses which isn't quite right. Unfortunately mapgen for cities just isn't intelligent enough to make specials relate directly to towns around them yet, but in the future we want to make that happen. Some day soon city generation is likely to get gutted and rebuilt from the ground up
